compiler/src/dotty/tools/dotc/transform/localopt/FormatChecker.scala

```diff
@@ -209,17 +215,20 @@ class TypedFormatChecker(partsElems: List[Tree], parts: List[String], args: List
     def accepts(arg: Type): Boolean =
       kind match
         case BooleanXn  => arg == defn.BooleanType orElse warningAt(CC)("Boolean format is null test for non-Boolean")
```

Comment:
I know that you didn't change this line, but could you make the warning message more explicit?
I think that it is unnecessarily terse, which makes it hard to understand.